hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -91,16 +90,8 @@ boolean contains(String name) {
      * @param result The ProxyInfo of all proxies in this registry.
      */
     public void getProxyInfos(Collection<ProxyInfo> result) {
-        for (Map.Entry<String, DistributedObjectFuture> entry : proxies.entrySet()) {
-            final DistributedObjectFuture future = entry.getValue();
-            if (!future.isSet()) {
```

Comment:
_Scenario_: You need to do an operation only one times in a cluster during proxy creation, if a new node joins during proxy creation, that new node should not run that operation again. Loading data from a store may be an example of this, you don't want every joining node to load data again and again.

For this scenario, removing `future.isSet()` may be problematic?